lib/jekyll-webp/webpGenerator.rb

```diff
@@ -69,6 +69,12 @@ def generate(site)
               FileUtils::mkdir_p(imgdir_destination + imgfile_relative_path)
               outfile_fullpath_webp = File.join(imgdir_destination + imgfile_relative_path, outfile_filename)
 
+              # Keep the webp file from being cleaned by Jekyll
+              site.static_files << WebpFile.new(site,
```

Comment:
Please don't write the static file to the site unless it was successfully created. So make sure the sequence is as it was before, first run the WebpExec and then do the WebpFile.new. Otherwise if there is an error then Jekyll thinks there exists a webp file that infact was never written :)

Reply:
Good point!